README.md

```diff
@@ -39,5 +39,26 @@ gem 'method_man', require: 'method_object'
     last_name: 'Smith',
     message: 'Hi',
   )
-  => ["John Smith", 'Hi', 42]
+  => ['John Smith', 'Hi', 42]
 ```
+
+Also allows automatic delegation inspired by Go's
+[delegation](https://nathany.com/good/).
+
+
+```ruby
+  require 'method_object'
+
+  class MakeArbitraryArray < MethodObject
+    attrs(:company)
+
+    def call
+      [company.name, company_name]
```

Comment:
At first glance I missed that the second method call show the delegation feature you implemented. It would be nice to make this standout more. How do you feel about switching the order so that `company_name` is first?  Another option is to put each item on a line and include a comment. 

```ruby
def call
  [
    company_name, # method `name` is delegated to the company
    ect....
  ]
end
```